app/config/services.yml

```diff
@@ -30,18 +30,17 @@ services:
         arguments:
             $emailSender: '%app.notifications.email_sender%'
 
+    # When your service definition only requires arguments to be specified,
+    # you can use a shorter syntax for these:
     AppBundle\Twig\AppExtension:
-        arguments:
-            $locales: '%app_locales%'
+        $locales: '%app_locales%'
 
-    AppBundle\EventListener\CommentNotificationSubscriber:
-        arguments:
-            $sender: '%app.notifications.email_sender%'
+    # The short syntax can also be inlined:
+    AppBundle\EventListener\CommentNotificationSubscriber: { $sender: '%app.notifications.email_sender%' }
 
     AppBundle\EventListener\RedirectToPreferredLocaleSubscriber:
-        arguments:
-            $locales: '%app_locales%'
-            $defaultLocale: '%locale%'
+        $locales: '%app_locales%'
+        $defaultLocale: '%locale%'
 
     # needed for the localizeddate Twig filter
-    Twig\Extensions\IntlExtension: ~
+    Twig\Extensions\IntlExtension: ~ # same as []
```

Comment:
You're right. `~` means `null` using yaml. 
But actually, configuring `null` for a service definition does not mean anything, and [is converted to an empty array behind the scene](https://github.com/symfony/symfony/pull/21214).

I felt like adding this comment would help the user to understand the service definition is either the full mapping of definition's attributes (`{ attributes: [...], tags: [...], public: false, ... }`) or directly the list (or mapping for named arguments) of constructor parameters.

But maybe that's not necessary.